Source/WebCore/platform/mediastream/gstreamer/PipeWireCaptureDeviceManager.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2025 Igalia S.L
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * aint with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "config.h"
+#include "PipeWireCaptureDeviceManager.h"
+
+#if ENABLE(MEDIA_STREAM) && USE(GSTREAMER)
+
+#include "GStreamerVideoCaptureSource.h"
+#include "MockRealtimeMediaSourceCenter.h"
+#include <wtf/Scope.h>
+
+namespace WebCore {
+
+GST_DEBUG_CATEGORY(webkit_pipewire_capture_device_manager_debug);
+#define GST_CAT_DEFAULT webkit_pipewire_capture_device_manager_debug
+
+RefPtr<PipeWireCaptureDeviceManager> PipeWireCaptureDeviceManager::create(OptionSet<CaptureDevice::DeviceType> deviceTypes)
+{
+    return adoptRef(*new PipeWireCaptureDeviceManager(deviceTypes));
+}
+
+PipeWireCaptureDeviceManager::PipeWireCaptureDeviceManager(OptionSet<CaptureDevice::DeviceType> deviceTypes)
+    : m_deviceTypes(deviceTypes)
+    , m_pipewireDeviceProvider(adoptGRef(gst_device_provider_factory_get_by_name("pipewiredeviceprovider")))
+{
+    static std::once_flag onceFlag;
+    std::call_once(onceFlag, [] {
+        GST_DEBUG_CATEGORY_INIT(webkit_pipewire_capture_device_manager_debug, "webkitpipewirecapturedevicemanager", 0, "WebKit PipeWire Capture Device Manager");
+    });
+}
+
+void PipeWireCaptureDeviceManager::computeCaptureDevices()
+{
+    if (MockRealtimeMediaSourceCenter::mockRealtimeMediaSourceCenterEnabled())
+        return;
+
+    if (!m_pipewireDeviceProvider || !gstObjectHasProperty(GST_OBJECT_CAST(m_pipewireDeviceProvider.get()), "fd")) {
```

Comment:
Isn't `gstObjectHasProperty` requiring an ASCIILiteral already? Wouldn't you need `"fd"_s` here?